tasks/src/main/java/ru/sberbank/school/task02/calculator/CurrencyCalculator.java

```diff
@@ -0,0 +1,61 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.FxConversionService;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.NonNull;
+
+public class CurrencyCalculator implements FxConversionService {
+    final ExternalQuotesService externalQuotesService;
+
+    public CurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(@NonNull ClientOperation operation,
+                              @NonNull Symbol symbol,
+                              @NonNull BigDecimal amount) {
+        System.out.println("Get request volume: " +  amount );
+        return operation(operation, symbol, amount);
+    }
+
+    private BigDecimal operation(ClientOperation operation, Symbol symbol, BigDecimal amount) {
+        List<Quote> quotes = filterQutesList(externalQuotesService.getQuotes(symbol), amount);
+        if (quotes.size() > 0 && operation == ClientOperation.SELL) {
+            return quotes.get(0).getBid();
+        }
+        if (quotes.size() > 0 && operation == ClientOperation.BUY) {
+            return quotes.get(0).getOffer();
+        }
+        return BigDecimal.ZERO;
+    }
+
+    private List<Quote> filterQutesList(List<Quote> quotes, BigDecimal amountOfRequest) {
+        showQuotes(quotes);
+        quotes.sort(new CompareQutes());
+        List<Quote> filteredList = new ArrayList<>();
+        for (Quote quote : quotes) {
+            if (quote.getVolumeSize().compareTo(amountOfRequest) > 0
+            || quote.getVolume().isInfinity()) {
```

Comment:
&#39;if&#39; child has incorrect indentation level 12, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-576046954>*